### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by Lukellmann**
* [Kotlin](https://github.com/JetBrains/kotlin) 1.710 -> 1.7.20
* [KSP](https://github.com/google/ksp) 1.7.10-1.0.6 -> 1.7.20-1.0.6
* [Ktor](https://github.com/ktorio/ktor) 2.1.1 -> 2.1.2
* [kotlinx.serialization](https://github.com/Kotlin/kotlinx.serialization) 1.3.3 -> 1.4.0
* [Binary compatibility validator](https://github.com/Kotlin/binary-compatibility-validator) 0.11.0 -> 0.11.1
* [JUnit 5](https://github.com/junit-team/junit5) 5.9.0 -> 5.9.1
* [MockK](https://github.com/mockk/mockk) 1.12.5 -> 1.13.2

### CODE REVIEW

**Comment by Lukellmann** [comment]
The removals in apidumps are probably because of [this](https://kotlinlang.org/docs/whatsnew1720.html#more-optimized-cases-of-delegated-properties), but should be fine as they unlikely to be used by kord users (tbh I don't really know what they did in the first place).

File: voice/src/main/kotlin/streams/DefaultStreams.kt [hunk 1]
```diff
@@ -84,7 +83,7 @@ public class DefaultStreams(
     private val _ssrcToUser: AtomicRef<MutableMap<UInt, Snowflake>> =
         atomic(mutableMapOf())
 
-    override val ssrcToUser: Map<UInt, Snowflake> by _ssrcToUser
+    override val ssrcToUser: Map<UInt, Snowflake> get() = _ssrcToUser.value
```
Comment (Lukellmann) [inline voice/src/main/kotlin/streams/DefaultStreams.kt:86]: this produced a compiler error, so I had to change it to manual delegation

**Review by NoComment1105 — APPROVED** [review]

**Comment by Lukellmann** [comment]
Now also included Ktor 2.1.2 that was released today

**Comment by Lukellmann** [comment]
Oops, forgot Binary compatibility validator

**Pull request merged**